f8a_worker/workers/git_stats.py

```diff
@@ -0,0 +1,159 @@
+    def _get_trend_str(trend):
+        """Convert numerical representation of trend to its string representation.
+
+        :param trend: numerical representation of thrend
+        :return: textual representation of trend
+        """
+        if trend > 3:
+            return 'increasing'
+        elif trend > -3:
+            return 'calm'
+        else:
+            return 'decreasing'
+
+    @staticmethod
+    def _get_orgs(log):
+        """Aggregate all organizations that were found in the log.
+
+        :param log: log on which organizations should be found
+        :return: a list of found organizations in the log
+        """
+        orgs = set()
+
+        for entry in log:
+            email = entry['author']['email'].split('@')
+            if len(email) == 2:
+                orgs.add(email[1])
+
+        return list(orgs)
+
+    @classmethod
+    def _compute_stats(cls, log, starting_date):
+        """Compute statisics on log.
+
+        :param log: log entries on which statistics should be computed
+        :param starting_date: starting date of log
+        :return: computed statistics
+        """
+        trend = cls._get_trend(log, starting_date) if log else None
+        return {
+            'commit_count': len(log),
+            'committer_count': len(list(set(item['author']['email'] for item in log))),
+            'oldest_commit': log[-1]['author']['date'] if log else None,
+            'newest_commit': log[0]['author']['date'] if log else None,
+            'average_changes': cls._get_average_changes(log) if log else None,
+            'trend': trend,
+            'trend_status': cls._get_trend_str(trend)
+        }
+
+    def execute(self, arguments):
+        self._strict_assert('ecosystem' in arguments)
+        self._strict_assert('name' in arguments)
+        self._strict_assert('url' in arguments)
+
+        master_log = self._get_log(arguments['url'])
+        if not master_log:
+            raise ValueError("No log for master branch to inspect")
+
+        now = time()
+        last_year_timestamp = now - timedelta(days=self._DAYS_PER_YEAR).total_seconds()
+        last_month_timestamp = now - timedelta(days=self._DAYS_PER_MONTH).total_seconds()
+
+        last_year_log = [item for item in master_log if item['author']['date'] >= last_year_timestamp]
+        last_month_log = [item for item in master_log if item['author']['date'] >= last_month_timestamp]
+
+        master_stats = {}
+        records = (
+            ('overall', master_log, master_log[-1]['author']['date'] if master_log else None),
+            ('year', last_year_log, last_year_timestamp),
+            ('month', last_month_log, last_month_timestamp)
+        )
+        for key, log, starting_date in records:
+            master_stats[key] = self._compute_stats(log, starting_date)
+
+        orgs = self._get_orgs(master_log)
+
+        return {'summary': [], 'status': 'success', 'details': {'master': master_stats, 'organizations': orgs}}
```

Comment:
Why do we have the organizations separately ? What about having it in `master_stats` ?